torchrl/envs/utils.py

```diff
@@ -181,28 +188,25 @@ def step_mdp(
             next_tensordict.update(out)
             return next_tensordict
         return out
+
     out = tensordict.get("next").clone(False)
-    excluded = set()
+    excluded = {action_key}
     if exclude_done:
         excluded.add(done_key)
     if exclude_reward:
         excluded.add(reward_key)
     if len(excluded):
         out = out.exclude(*excluded, inplace=True)
-    td_keys = None
+
+    if not exclude_action:
+        out._set(action_key, tensordict.get(action_key))
```

Comment:
Here if action has to be kepts we're removing it and adding it back, which is expensive, let's avoid that

Reply:
we are not removing it, the exclusion before is on "next"